### PR TITLE
V13: login screen design refresh

### PR DESCRIPTION
**Pull request by JesmoDev**
### Description

A refresh of the login screen that implements the latest brand colors of Umbraco.

### Screenshots

image (9)
Screenshot 2023-11-16 at 09 13 57
image (11)

### CODE REVIEW

File: src/Umbraco.Web.UI.Login/src/components/pages/mfa.page.element.ts [hunk 1]
```diff
@@ -1,243 +1,279 @@
...
+        }
+        this.buttonState = 'failed';
+        return;
+      }
+
+      this.buttonState = 'success';
+
+      const returnPath = umbAuthContext.returnPath;
+      if (returnPath) {
+        location.href = returnPath;
```
[the remark this answers is not in the document]
Comment (iOvergaard) [reply src/Umbraco.Web.UI.Login/src/components/pages/mfa.page.element.ts:108]: This has been amended in cf0697a

[the remark this answers is not in the document]
Comment (iOvergaard) [reply src/Umbraco.Web.UI.Login/src/components/pages/mfa.page.element.ts:108]: This has been amended in cf0697a

**Comment by nielslyngsoe** [comment]
@JesmoDev @iOvergaard 

A few thoughts of mine, that you can consider (Just based on the screenshots)

I think 'Or' should be lowercase, as it is in relation to what was placed above. It's a bit like a sentence and in that perspective, it feels off with capitalized O.

I see some 'greyed-out' fields in the two last screenshots. The dropdown and the username field. This makes them look disabled. Could we avoid that?

Is it the UUI Dropdown(or what the name is) that is used for the authentication method? It's not the same height as the input fields and I would like it to be so, if possible. (if UUI, then we should correct it there as well. )

It looks really good, looking forward to it :-)

**Comment by iOvergaard** [comment]
@nielslyngsoe good idea with the lowercase 'or'. I just updated it.

The other thing you are pointing out is a normal uui-select, which I see has `background-color: transparent` on the inner `<select>` element on UUI:

image

We'll fix it locally here, but should probably be fixed in uui.

**Comment by iOvergaard** [comment]
The screenshots have been updated to reflect the changes.

**Comment by bjarnef** [comment]
I wonder is any specific thoughts behind the rounded buttons, because it doesn't align with the backoffice buttons or the input fields :) 

I think the buttons for the external login providers look like input fields, because the look it very similar to the input fields. So any not familiar to Umbraco may think these are inputs.

**Comment by iOvergaard** [comment]
@bjarnef 

> I wonder is any specific thoughts behind the rounded buttons, because it doesn't align with the backoffice buttons or the input fields :)

Moving forward, the design of the login screen will have its' own identity which may not look exactly like the Backoffice. This identity will follow the brand guidelines of Umbraco and that is why we are seeing more rounded corners and slightly different colors.

The new color codes will eventually land in the UI library as well as the Backoffice. It's mainly the primary blue color that's changed.

> I think the buttons for the external login providers look like input fields, because the look it very similar to the input fields.

I agree and can see it now that you mention it. I will bring it up with our design team to discuss what can be done to rectify this. Thanks!

**Comment by bjarnef** [comment]
@iOvergaard a bit drop shadow on the button could enhance the button look, but with a flat design button may look less like buttons.
https://dribbble.com/shots/22925880-Login-Screen-For-E-Commerce-Product

Alternatively a secondary button color:
https://dribbble.com/shots/22808838-Login-Page-Web-Design-Concept-for-Minimalist-Furniture

I something else than white, which is typical background color on input fields.
https://dribbble.com/shots/22826180-Day-4-Login-screen

**Comment by bjarnef** [comment]
@iOvergaard okay, it will wait and see what happens 😁
I kinda like the minimalist example here https://dribbble.com/shots/22808838-Login-Page-Web-Design-Concept-for-Minimalist-Furniture with is a bit similar to the new design.

Not sure it the gray background will be too boring (or too simple) compared to the current login screen. Maybe a shade of the blue color or a complementary color would help. Just thinking out loud 😉

**Review by iOvergaard — APPROVED** [review]
LGTM

**Comment by AndersBrohus** [comment]
Hi :)

Is there any option to style the new login page other then using the LoginBackgroundImage and LoginLogoImage properties?

We would like to add a max width to an SVG and perhabs hide the rouned corners inside the image :)

**Comment by iOvergaard** [comment]
@AndersBrohus Yes there is, and more options will follow in the next RC. You can see the list here on [the Umbraco Docs](https://docs.umbraco.com/umbraco-cms/v/13.latest-rc/fundamentals/backoffice/login#custom-css).

**Comment by AndersBrohus** [comment]
Ohhh nice @iOvergaard 🥳🎉

Do you have an estimate on when the next RC will come? :)

**Comment by iOvergaard** [comment]
@AndersBrohus No date is set nor is another RC planned just yet. At the latest, Umbraco 13 will launch on December 14th.

**Comment by AndersBrohus** [comment]
Sounds great - We have upgraded a website to RC3 and are watching it, until now there is zero issues, so we are waiting to december for the final release :)

**Comment by DORE-PD** [comment]
I'm missing a "Show password" button here. Is there a way to add this? @JesmoDev

**Comment by bjarnef** [comment]
I agree with @DORE-PD that the "Show password" button is missing.

I think this was useful when using autofill or password manager. Of course we can inspect DOM and change input type from `password` to `text`.

**Comment by iOvergaard** [comment]
@DORE-PD @bjarnef I completely agree. Actually, it was planned to use the [UI library for the password field](https://uui.umbraco.com/?path=/docs/uui-input-password--docs), but just before the launch we decided against it since custom elements with Shadow DOM do not play nicely with password managers (#15075) for the time being.

Our hope is that password managers will eventually support shadow DOM and we can go back to the UI library, where we get nice things out of the box such as a "Show password" button. I know that Edge supports it directly on native password fields. Perhaps it will land in upstream Chromium at some point. Meanwhile, feel free to submit a pull request if you have some time to spend on the current login screen.

**Pull request merged**